src/wobjectcpp.h

```diff
@@ -0,0 +1,124 @@
+#pragma once
+#include "wobjectdefs.h"
+
+namespace w_cpp {
+
+using w_internal::StringView;
+using w_internal::viewLiteral;
+
+template<typename T>
+constexpr auto makeProperty(StringView name, StringView type) {
+    return w_internal::MetaPropertyInfo<T>{ name, type, {}, {}, {}, {}, {} };
+}
+
+#if __cplusplus <= 201700L
+template<bool... Bs>
+constexpr bool all() {
+    bool b = true;
+    w_internal::ordered2<bool>({(b = b && Bs)...});
+    return b;
+}
+#endif
+
+template<class F, int Flags, class IC = int, class ParamTypes = w_internal::StringViewArray<>, class ParamNames = w_internal::StringViewArray<>>
+struct MetaMethodInfoBuilder {
+    StringView name;
+    F func;
+    ParamTypes paramTypes{};
+    ParamNames paramNames{};
+    static constexpr int flags = Flags;
+    using IntegralConstant = IC;
+
+#if __cplusplus > 201700L
+    template<class... Args, class = std::enable_if_t<(std::is_same_v<std::decay_t<Args>, StringView> && ...)>>
+#else
+    template<class... Args, class = std::enable_if_t<all<std::is_same<std::decay_t<Args>, StringView>::value...>()>>
+#endif
+    constexpr auto setParamTypes(Args... paramTypes) const
+        -> MetaMethodInfoBuilder<F, Flags, IC, w_internal::StringViewArray<sizeof... (Args)>, ParamNames> {
+        return {name, func, {paramTypes...}, paramNames};
+    }
+#if __cplusplus > 201700L
+    template<class... Args, class = std::enable_if_t<(std::is_same_v<std::decay_t<Args>, StringView> && ...)>>
+#else
+    template<class... Args, class = std::enable_if_t<all<std::is_same<std::decay_t<Args>, StringView>::value...>()>>
+#endif
+    constexpr auto setParamNames(Args... paramNames) const
+        -> MetaMethodInfoBuilder<F, Flags, IC, ParamTypes, w_internal::StringViewArray<sizeof... (Args)>> {
+        return {name, func, paramTypes, {paramNames...}};
+    }
+    template<int... fs>
+    constexpr auto addFlags(w_internal::W_MethodFlags<fs>...) const

```

Comment:
Is the user supposed to reach into the internal namespace?

Reply:
That is not necessary.

[wobjectdefs.h#L185](https://github.com/woboq/verdigris/blob/master/src/wobjectdefs.h#L185) and follows, those flags are given public names like `W_Access::Private`

But the types are still `w_internal::W_MethodFlags`.